src/ray/protobuf/ray_client.proto

```diff
@@ -76,6 +76,10 @@ message ClientTask {
   TaskOptions options = 7;
   // Options passed to create the default remote task excution environment.
   TaskOptions baseline_options = 8;
+  // Thread that scheduled this task
+  int64 thread_id = 10;
+  // Identifier for this ticket
+  int32 req_id = 11;
```

Comment:
Is it possible to make these fields request metadata? The reason is that we may move a request type between unary and streaming RPCs, and these fields become obsolete if the request type is handled in streaming mode.

Reply:
An example is that I'm moving `ClientTask` to streaming RPC in #18298. In future we may move all KV.* operations to streaming. Also I will be moving `Get` to unary (but replay protection does not matter in that case).

Reply:
Hmm, my only concern there is that using metadata is considerably high overhead (for example it will encode the string "thread_id" and "req_id" into the request, instead of inferring the names of those values while marshalling and demarshalling). How risky do you think it would be to drop the fields entirely after they're moved to streaming RPCs?

Reply:
Decided overhead is fine, since we're already serializing the client_id in this manner, and will significantly simplify swapping over the unary RPCs